src/EditorFeatures/Core/ExternalAccess/IntelliCode/Api/IIntentSourceProvider.cs

```diff
@@ -77,21 +77,31 @@ public IntentRequestContext(string intentName, SnapshotSpan currentSnapshotSpan,
 
         /// <summary>
         /// The text changes that should be applied to the <see cref="IntentRequestContext.CurrentSnapshotSpan"/>
+        /// TODO - Remove once intellicode switches over to reading <see cref="DocumentChanges"/> instead.
         /// </summary>
+        [Obsolete("Use DocumentChanges instead")]
         public readonly ImmutableArray<TextChange> TextChanges { get; }
 
+        /// <summary>
+        /// The text changes that should be applied to each document.
+        /// </summary>
+        public readonly ImmutableDictionary<Uri, ImmutableArray<TextChange>> DocumentChanges;
```

Comment:
@pgroene you'll need to switch to using this.  Should be fairly self explanatory, contains the text changes that should be applied to each document

Reply:
why URIs instead of DocumentIds?